lib/cisco_node_utils/radius_server_group.rb

```diff
@@ -0,0 +1,110 @@
+# Radius Server Group provider class
+
+# Jonathan Tripathy et al., October 2015
+
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'node_util'
+
+module Cisco
+  # RadiusServerGroup - node utility class for
+  # Raidus Server configuration management
+  class RadiusServerGroup < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      unless name.is_a?(String)
+        fail ArgumentError, 'Invalid value (Name is not a String)'
+      end
+
+      @name = name
+
+      create if instantiate
+    end
+
+    def self.radius_server_groups
+      hash = {}
+      group_list = config_get('radius_server_group', 'group')
+      return hash if group_list.nil?
+
+      group_list.each do |id|
+        hash[id] = RadiusServerGroup.new(id, false)
+      end
+
+      hash
+    end
+
+    def create
+      config_set('radius_server_group',
+                 'group',
+                 state: '',
+                 name:  @name)
+    end
+
+    def destroy
+      config_set('radius_server_group',
+                 'group',
+                 state: 'no',
+                 name:  @name)
+    end
+
+    def ==(other)
+      name == other.name
+    end
+
+    def default_servers
+      val = config_get_default('radius_server_group', 'servers')
+      val
+    end
+
+    def servers
+      val = config_get('radius_server_group', 'servers', @name)
+      val = [] if val.nil?
```

Comment:
I'm not sure I agree with this. We shouldn't make the normal getters return the default value. I know it amounts to the same thing in this case (empty array), but didn't we move away from tying default values to actual configuration?